code/go/0chain.net/chaincore/chain/handler.go

```diff
@@ -57,10 +57,47 @@ func chainhandlersMap(c Chainer) map[string]func(http.ResponseWriter, *http.Requ
 	return m
 }
 
-func handlersMap(c Chainer) map[string]func(http.ResponseWriter, *http.Request) {
+func minerHandlersMap(c Chainer) map[string]func(http.ResponseWriter, *http.Request) {
 	transactionEntityMetadata := datastore.GetEntityMetadata("txn")
 	m := map[string]func(http.ResponseWriter, *http.Request){
+		"/": common.WithCORS(common.UserRateLimit(
+			HomePageAndNotFoundHandler,
+		)),
+		"/_diagnostics": common.UserRateLimit(
+			DiagnosticsHomepageHandler,
+		),
+		"/_diagnostics/current_mb_nodes": common.UserRateLimit(
+			DiagnosticsNodesHandler,
+		),
+		"/_diagnostics/dkg_process": common.UserRateLimit(
+			DiagnosticsDKGHandler,
+		),
+		"/_diagnostics/round_info": common.UserRateLimit(
+			RoundInfoHandler(c),
+		),
+		"/v1/transaction/put": common.WithCORS(common.UserRateLimit(
+			datastore.ToJSONEntityReqResponse(
+				datastore.DoAsyncEntityJSONHandler(
+					memorystore.WithConnectionEntityJSONHandler(PutTransaction, transactionEntityMetadata),
+					transaction.TransactionEntityChannel,
+				),
+				transactionEntityMetadata,
+			),
+		)),
+	}
+	if node.Self.Underlying().Type == node.NodeTypeMiner {
+		m[GetBlockV1Pattern] = common.UserRateLimit(
+			common.ToJSONResponse(
+				GetBlockHandler,
+			),
+		)
+	}
 
+	return m
+}
+
+func sharderHandlersMap(c Chainer) map[string]func(http.ResponseWriter, *http.Request) {
+	m := map[string]func(http.ResponseWriter, *http.Request){
```

Comment:
Sorry, I forget to mention that the endpoints of `/estimate_txn_fee` and `fees_table` endpoints are all miners endpoints, so as some other endpoints, such as `/get/latest_finalized` and `/latest_finalized_magic_block_summary`, etc. Basically all endpoints read data from MPT or data struct other than event db should be supported by both miners and sharders. We'd better keep them in common place of handlersMap and only move specific handlers to minerHandlerMap and sharderHandlersMap.